changes/2489.fixed

```diff
@@ -0,0 +1 @@
+Fixed custom field regex validation always anchored to beginning of string.
```

Comment:
This is arguably a behavior change rather than a bug fix. It's going to make existing regex rules less strict?

Reply:
I would argue that it was broken since that behavior wasn't documented anywhere and the form validator uses `re.search`. I guess it could have been worked around by starting the regex with `.*` and this change wouldn't break that workaround.

Reply:
Happy to discuss though

Reply:
I didn't know that about the form field validator. In that case I'm fine with treating this as a bug fix.